cardano-chain-gen/src/Cardano/Mock/Forging/Tx/Conway.hs

```diff
@@ -124,6 +184,81 @@ mkSimpleTx isValid' txBody =
     , auxiliaryData = maybeToStrictMaybe Nothing
     }
 
+mkSimpleDCertTx ::
+  [(StakeIndex, StakeCredential StandardCrypto -> ConwayTxCert StandardConway)] ->
+  ConwayLedgerState ->
+  Either ForgingError (AlonzoTx StandardConway)
+mkSimpleDCertTx consDCert st = do
+  dcerts <- forM consDCert $ \(stakeIndex, mkDCert) -> do
+    cred <- resolveStakeCreds stakeIndex st
+    pure (mkDCert cred)
+  mkDCertTx dcerts (Withdrawals mempty) Nothing
+
+mkDepositTxPools ::
+  ConwayUTxOIndex ->
+  Integer ->
+  ConwayLedgerState ->
+  Either ForgingError (AlonzoTx StandardConway)
+mkDepositTxPools inputIndex deposit state' = do
+  (inputPair, _) <- resolveUTxOIndex inputIndex state'
+
+  let input = Set.singleton (fst inputPair)
+      BabbageTxOut addr' (MaryValue inputValue _) _ _ = snd inputPair
+      change =
+        BabbageTxOut
+          addr'
+          (valueFromList (fromIntegral $ fromIntegral inputValue - deposit) [])
+          NoDatum
+          SNothing
+
+  pure $
+    mkSimpleTx True $
+      consTxBody
+        input
+        mempty
+        mempty
+        (StrictSeq.fromList [change])
+        SNothing
+        (Coin 0)
+        mempty
+        (allPoolStakeCert' state')
+        (Withdrawals mempty)
+
+mkDummyRegisterTx :: Int -> Int -> Either ForgingError (AlonzoTx StandardConway)
+mkDummyRegisterTx n m = mkDCertTx consDelegCert (Withdrawals mempty) Nothing
+  where
+    consDelegCert =
+      mkRegTxCert SNothing
+        . KeyHashObj
+        . KeyHash
+        . mkDummyHash (Proxy @(ADDRHASH StandardCrypto))
+        . fromIntegral
+        <$> [n, m]
+
+mkRegTxCert ::
+  StrictMaybe Coin ->
+  StakeCredential StandardCrypto ->
+  ConwayTxCert StandardConway
+mkRegTxCert coin' = mkTxDelegCert (`ConwayRegCert` coin')
+
+mkUnRegTxCert ::
+  StrictMaybe Coin ->
+  StakeCredential StandardCrypto ->
+  ConwayTxCert StandardConway
+mkUnRegTxCert coin' = mkTxDelegCert (`ConwayUnRegCert` coin')
+
+mkDelegTxCert ::
+  Delegatee StandardCrypto ->
+  StakeCredential StandardCrypto ->
+  ConwayTxCert StandardConway
+mkDelegTxCert delegatee = mkTxDelegCert (`ConwayDelegCert` delegatee)
+
+mkTxDelegCert ::
+  (StakeCredential StandardCrypto -> ConwayDelegCert StandardCrypto) ->
+  StakeCredential StandardCrypto ->
+  ConwayTxCert StandardConway
+mkTxDelegCert = (ConwayTxCertDeleg .)
```

Comment:
minor change if you agree, just wondering if this might be a little too point free?

Reply:
I went back and forth a few times, I don't mind making it a little more readable

Reply:
no big deal we can leave it 👍